File: algorithms/graph/planarity-testing/csharp/PlanarityTesting.cs
using System;
using System.Collections.Generic;

public class PlanarityTesting
{
    public static int Solve(int[] arr)
    {
        int n = arr[0], m = arr[1];
        var edges = new HashSet<long>();
        for (int i = 0; i < m; i++)
        {
            int u = arr[2+2*i], v = arr[2+2*i+1];
            if (u != v)
            {
                int a = Math.Min(u, v), b = Math.Max(u, v);
                edges.Add((long)a * n + b);
            }
        }
        int e = edges.Count;
        if (n < 3) return 1;
        return e <= 3 * n - 6 ? 1 : 0;
    }
}

File: algorithms/graph/planarity-testing/rust/planarity_testing.rs
use std::collections::HashSet;

pub fn planarity_testing(arr: &[i32]) -> i32 {
    let n = arr[0] as usize;
    let m = arr[1] as usize;
    let mut edges = HashSet::new();
    for i in 0..m {
        let u = arr[2 + 2 * i] as usize;
        let v = arr[2 + 2 * i + 1] as usize;
        if u != v {
            let a = u.min(v);
            let b = u.max(v);
            edges.insert((a, b));
        }
    }
    let e = edges.len();
    if n < 3 { return 1; }
    if e <= 3 * n - 6 { 1 } else { 0 }
}
